sympy/logic/boolalg.py

```diff
@@ -120,12 +121,18 @@ def equals(self, other):
         False
 
         """
-        from sympy.logic.inference import satisfiable
-        from sympy.core.relational import Relational
-
-        if self.has(Relational) or other.has(Relational):
-            raise NotImplementedError('handling of relationals')
-        return self.atoms() == other.atoms() and \
+        from sympy.core.symbol import Symbol
+
+        def ok(f):
+            return isinstance(f, (BooleanFunction, Symbol)) and (not f.args or
+                all(ok(a) for a in f.args))
+        if not ok(self) or not ok(other):
+            raise NotImplementedError('non-literal BooleanFunction')
+        more = self.atoms()
+        less = other.atoms()
+        if len(more) < len(less):
+            more, less = less, more
+        return not less - more and \
             not satisfiable(Not(Equivalent(self, other)))
```

Comment:
I don't understand what the length check is doing here.

Reply:
The atoms of one expression must be a subset of the other. If they are and both expressions have the same truth table then they are considered equivalent because other symbols were extraneous.

Reply:
Can you give an example of extraneous symbols? Why can't both expressions have extraneous symbols?

Reply:
`y` is redundant in `x&y|x&~y`. You are right, though, both expressions could have extraneous symbols. Perhaps it is just better to do simplification that would remove such symbols.